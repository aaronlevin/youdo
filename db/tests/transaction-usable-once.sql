\i create.sql
\unset ON_ERROR_STOP

\echo insert without transaction
begin;
insert into yd_user (name) values ('Charlie');
end;

\echo update without transaction
begin;
update yd_user set name='Bob' where id = 0;
end;

\echo delete without transaction
begin;
delete from yd_user where id = 0;
end;

\echo insert with old transaction
begin;
insert into transaction (yd_userid, yd_ipaddr, yd_useragent)
values (0, '127.0.0.1', :'script')
returning id
    \gset txn
end;
begin;
insert into yd_user (txnid, name) values (:txnid, 'Charlie');
end;

\echo update with old transaction
begin;
insert into transaction (yd_userid, yd_ipaddr, yd_useragent)
values (0, '127.0.0.1', :'script')
returning id
    \gset txn
end;
begin;
update yd_user set txnid=:txnid, name='Charlie' where id=0;
end;
