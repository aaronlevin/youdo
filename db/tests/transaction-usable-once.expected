insert without transaction
psql:: ERROR:  No open transaction
update without transaction
psql:: ERROR:  No open transaction
delete without transaction
psql:: ERROR:  No open transaction
insert with old transaction
psql:: ERROR:  No open transaction
update with old transaction
psql:: ERROR:  No open transaction
